src/index.spec.ts

```diff
@@ -545,5 +548,37 @@ describe('EppoJSClient E2E test', () => {
       // Assignments now working
       expect(client.getStringAssignment('subject', flagKey)).toBe('control');
     });
+
+    describe('With reloaded index module', () => {
+      // eslint-disable-next-line @typescript-eslint/ban-types
+      let init: Function;
+      // eslint-disable-next-line @typescript-eslint/ban-types
+      let getInstance: Function;
+      beforeEach(() => {
+        jest.isolateModules(() => {
+          // Isolate and re-require so that the static instance is reset to it's default state
+          // eslint-disable-next-line @typescript-eslint/no-var-requires
+          const reloadedModule = require('./index');
+          init = reloadedModule.init;
+          getInstance = reloadedModule.getInstance;
+        });
```

Comment:
This `isolateModules()` and `require()` combo to reset the static EppoJSClient instance took me a bit to figure out and it had weird side effects, such as making test double (`td`) not work (hence me using `returnRac()` instead of mocking a response like other tests cases). Also, using this outside of a `beforeEach()`--which I did at first because I just have one test case--results in `console.log()` not actually logging anything 🤯 

TL;DR; use singletons and `jest.isolateModules()` with extreme caution, but it is a potentially useful tool to have your toolbox.